data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -226,10 +226,16 @@ private CloseableIterable<T> applyPosDeletes(CloseableIterable<T> records) {
 
     // if there are fewer deletes than a reasonable number to keep in memory, use a set
     if (posDeletes.stream().mapToLong(DeleteFile::recordCount).sum() < setFilterThreshold) {
-      return Deletes.filter(records, this::pos, Deletes.toPositionIndex(filePath, deletes));
+      PositionDeleteIndex positionIndex = Deletes.toPositionIndex(filePath, deletes);
+      Predicate<T> isInDeleteSet = record -> positionIndex.isDeleted(pos(record));
+      return hasColumnIsDeleted ?
+          Deletes.markDeleted(records, isInDeleteSet, this::markRowDeleted) :
```

Comment:
Just personal preference, but maybe we can push the pos function call down to Deletes (as we do it for streaming, but not the non-streaming)


```
PositionDeleteIndex positionDeleteIndex = Deletes.deletePositions(filePath, deletes);
if (posDeletes.stream().mapToLong(DeleteFile::recordCount).sum() < setFilterThreshold) {
   return hasColumnIsDeleted ?
      Deletes.markDeleted(records, this::pos, positionDeleteIndex, this::markDeleted) : 
      Deletes.filterDeleted(records, this::pos, positionDeleteIndex);
}

where Deletes.markDeleted is just a wrapper that makes the Predicate.

```

Reply:
`Deletes.markDeleted` is used by eq delete as well. I'd prefer to keep it as is. I made a new method `Deletes.filterDeleted` to make it more symmetric.

Reply:
Hm i didnt see markDeleted used anywhere else in code, maybe I missed it?  

I thought it would be more symmetric to have Deletes class take in PositionDeleteIndex (given it returns a PositionDeleteIndex) and Function<T, Long> rowToPosition (given other methods take it), but just style preference.